src/node_buffer.cc

```diff
@@ -907,33 +938,62 @@ void IndexOfBuffer(const FunctionCallbackInfo<Value>& args) {
   ASSERT(args[1]->IsObject());
   ASSERT(args[2]->IsNumber());
 
+  enum encoding enc = ParseEncoding(args.GetIsolate(),
+                                    args[3].As<String>(),
```

Comment:
`args[3]` must be a string before doing this operation. Add the `CHECK(args[3]->IsString())` above.